app/src/main/java/ai/brokk/project/MainProject.java

```diff
@@ -2240,6 +2240,58 @@ public DependencyUpdateScheduler getDependencyUpdateScheduler() {
         return dependencyUpdateScheduler;
     }
 
+    @Override
+    public boolean isEmptyProject() {
+        if (!hasGit()) {
+            return false; // Non-git projects are not considered "empty" in this sense
+        }
+
+        Set<ProjectFile> trackedFiles = repo.getTrackedFiles();
+
+        if (trackedFiles.isEmpty()) {
+            return true; // No files at all
+        }
+
+        // Check if all tracked files are configuration/metadata files
+        for (ProjectFile file : trackedFiles) {
```

Comment:
Also, only checking `trackedFiles` is maybe a bit too limiting for new projects - there may be a lot of scaffolding involved before committing initial files